templates/vue/src/components/node-modal/content-form/Icon.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div
+    class="icon"
+    :class="[`icon--${name}`, `icon--${size}`, { 'has-align-fix': fixAlign }]"
+  >
+    <img :src="iconSource" :alt="`${name}`" />
+  </div>
+</template>
+
+<script>
+import bold from "../../../assets/icons/bold.svg"
+import italic from "../../../assets/icons/italic.svg"
+import underline from "../../../assets/icons/underline.svg"
+import strike from "../../../assets/icons/strike.svg"
+import paragraph from "../../../assets/icons/paragraph.svg"
+import ul from "../../../assets/icons/ul.svg"
+import ol from "../../../assets/icons/ol.svg"
+import quote from "../../../assets/icons/quote.svg"
+import code from "../../../assets/icons/code.svg"
+import hr from "../../../assets/icons/hr.svg"
+import undo from "../../../assets/icons/undo.svg"
+import redo from "../../../assets/icons/redo.svg"
+
+export default {
+  props: {
+    name: {
+      type: String,
+      default: "",
+    },
+    size: {
+      type: String,
+      default: "normal",
+    },
+    modifier: {
+      type: Object,
+      default: null,
+    },
+    fixAlign: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    iconSource() {
+      let lookup = new Map()
+      let icons = [
+        bold,
+        italic,
+        underline,
+        strike,
+        code,
+        paragraph,
+        ul,
+        ol,
+        quote,
+        hr,
+        undo,
+        redo,
+      ]
+      let iconsStrings = [
+        "bold",
+        "italic",
+        "underline",
+        "strike",
+        "code",
+        "paragraph",
+        "ul",
+        "ol",
+        "quote",
+        "hr",
+        "undo",
+        "redo",
+      ]
+      for (let i in icons) {
+        lookup.set(iconsStrings[i], icons[i])
+      }
+      let usedIcon = lookup.get(this.name)
```

Comment:
I notice the icons and the strings are all the same, so we can just do:

```suggestion
      let usedIcon = this.name
```

Reply:
in the icon array, those aren't strings, those are references to strings with the full url as provided by webpack (with the ./dist..... and stuff), which is why the map is used to translate between the two.

Reply:
Ah good point, nevertheless I think we can shorten this to something like:

```
const lookup = {
  bold,
  italic,
  underline,
  ...etc
}
const usedIcon = lookup[this.name]
```

Reply:
Does it really work without the Map?

Reply:
Yeah, javascript objects are functionally identical to Maps if the keys are strings or numbers

Reply:
Still confused

`const lookup = {
  bold,
  italic,
  underline,
  ...etc
}
const usedIcon = lookup[this.name]
`

Will this not become 

`const lookup = {
  wpaddress/src/dist/bold7098709871234.svg,
  wpaddress/src/dist/italic7098709871234.svg,
  wpaddress/src/dist/underline7098709871234.svg,
  ...etc
}
const usedIcon = lookup[this.name]`

after compilation?

Reply:
No it's going to compile to:

```
const lookup = {
  bold: "wpaddress/src/dist/bold7098709871234.svg",
  italic: ".....",
  etc
}
```

It's called "object shorthand notation": https://alligator.io/js/object-property-shorthand-es6/

Reply:
Added!